Fix FQDN compliancy

To allow the dhcp hook script to set the correct entry as well, this patch creates
a special entry in /etc/hosts that the hook script will set to the right address.

Patch by Gordon Bos

--- a/controllers/InterfaceController.cpp
+++ b/controllers/InterfaceController.cpp
@@ -520,7 +520,7 @@
 		string hostname=FileUtils::GetContentAsString("/proc/sys/kernel/hostname");
 
 		// add self reference to fulfil FQDN requirement
-		string ip = "127.0.0.1";
+		string ip = "256.0.0.1";
 		Hosts::Entries e=h.Find(hostname);
 		Hosts::UpdateIP(e,ip,hostname);
 
--- a/utils/Hosts.cpp
+++ b/utils/Hosts.cpp
@@ -67,8 +67,8 @@
 	if(ip!=""){
 		Hosts::Entry ent(3);
 		ent[0]=ip;
-		ent[1]=name;
-		ent[2]=name+".localdomain";
+		ent[1]=name+".localdomain";
+		ent[2]=name;
 		e.push_back(ent);
 	}
 
@@ -99,8 +99,8 @@
 
 		for(Hosts::Entries::iterator entry=e.begin();entry!=e.end();entry++){
 			(*entry).erase((*entry).begin()+1,(*entry).end());
-			(*entry).insert((*entry).end(),name);
 			(*entry).insert((*entry).end(),name+".localdomain");
+			(*entry).insert((*entry).end(),name);
 		}
 
 	}
